providers/implementations/signature/rsa.c

```diff
@@ -277,18 +238,27 @@ static int rsa_setup_mgf1_md(PROV_RSA_CTX *ctx, const char *mdname,
                              const char *mdprops)
 {
     size_t len;
+    EVP_MD *md = NULL;
 
     if (mdprops == NULL)
         mdprops = ctx->propq;
 
     if (ctx->mgf1_mdname[0] != '\0')
         EVP_MD_free(ctx->mgf1_md);
 
-    if ((ctx->mgf1_md = EVP_MD_fetch(ctx->libctx, mdname, mdprops)) == NULL) {
+    if ((md = EVP_MD_fetch(ctx->libctx, mdname, mdprops)) == NULL) {
         ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_DIGEST,
                        "%s could not be fetched", mdname);
         return 0;
     }
+    /* The default for mgf1 is SHA1 - so allow SHA1 */
```

Comment:
That comment doesn't seem to go with what the following code is doing.

Reply:
The last param is sha_allowed.. So setting it to 1 allows sha1